mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -283,17 +319,25 @@ import scala.util.{Failure, Success}
 
     timer =>
       if (!timer.isTimerActive(ReceiveConnAck))
-        timer.startSingleTimer(ReceiveConnAck, ReceiveConnAckTimeout, data.settings.receiveConnAckTimeout)
-
+        timer.startSingleTimer(ReceiveConnAck,
+                               ReceiveConnAckTimeout(data.connectionId),
+                               data.settings.receiveConnAckTimeout)
       Behaviors
         .receivePartial[Event] {
-          case (context, ConnAckReceivedFromRemote(connAck, local))
+          case (context, connect @ ConnectReceivedLocally(connectionId, _, _, _))
+              if connectionId != data.connectionId =>
+            context.self ! connect
+            disconnect(context, data.remote, data)
+          case (_, event) if event.connectionId.nonEmpty && event.connectionId != data.connectionId =>
+            Behaviors.same
```

Comment:
This case is responsible for letting empty (thus session bound) or matching connection ids fall through to other cases.